### DISCUSSION TITLE
How to debug Turborepo pipeline

### DISCUSSION THREAD

**Discussion by durre** [vercel/turborepo#959] [Help] [answered]
I wanted to try Turborepo at a new company I just started working for. They have an existing monorepo based on yarn workspaces. When I execute `yarn turbo run build -vv --concurrency=1` I can see it starting to build packages, but when it reaches a certain package I just stops (have to use CTRL + C), without any indication why. This is the last of the output:

```
@mycompany/scylla-client:build: cache miss, executing e092187dca55cd5a
@mycompany/scylla-client:build: $ tsc
2022-03-27T16:42:11.662+0200 [DEBUG] run.@mycompany/scylla-client:build: caching output: outputs=[".turbo/turbo-build.log", "dist/**/*", "build/**/*"]
2022-03-27T16:42:11.662+0200 [DEBUG] run.@mycompany/scylla-client:build: done: status=complete duration=3.727558417s
```

The turbo-build.log looks OK to me, the built files are in the `dist/` folder. Any tips on how to proceed debugging this?

**Comment by durre** [comment]
By removing the packages one by one I managed to figure it out. There was a circular dependency in the package.json between two of the packages. Once removed, turborepo could proceed to build all the packages. The last package in the output was not one of the packages with a circular dependency so I think the original question still stands. Is there a way I could have gotten to this conclusion sooner with some better way of debugging?

**Comment by GauBen** [comment]
Hey there, I just ran into this very issue today, and I found the problem with `yarn turbo run build --graph`.
However, a solver should be baked-in instead of just blocking the pipeline.

**Comment by jaredpalmer** [comment, marked as answer]
In addition to `--graph`, you can also use the new `--dry` and `--dry=json` flags in `turbo` v1.2+. See docs here: https://turborepo.org/docs/reference/command-line-reference#--dry----dry-run
